### PR TITLE
Multi platform docker build

### LINKED ISSUE #152: Crashes on Apple Silicon
**Describe the bug**
On Apple Silicon, Chromium will crash on launch with the following message:

```
app_1 | Error: Protocol error (Target.setDiscoverTargets): Target closed.

app_1 | at /usr/src/app/node_modules/puppeteer-core/lib/cjs/puppeteer/common/Connection.js:71:63

app_1 | at new Promise (<anonymous>)

app_1 | at Connection.send (/usr/src/app/node_modules/puppeteer-core/lib/cjs/puppeteer/common/Connection.js:70:16)

app_1 | at Function.create (/usr/src/app/node_modules/puppeteer-core/lib/cjs/puppeteer/common/Browser.js:95:26)

app_1 | at ChromeLauncher.launch (/usr/src/app/node_modules/puppeteer-core/lib/cjs/puppeteer/node/Launcher.js:101:56)

app_1 | at processTicksAndRejections (internal/process/task_queues.js:93:5)
```

**To Reproduce**

1. Execute `$ docker-compose up --build -d`
2. See error

**What is the expected behavior?**
Chrome not to crash

**What is the actual behavior?**
Chrome crashes

**Possible solution**
Not sure –– hoping someone here might know what's going on

**Logs**
If applicable, add logs to help explain your problem.

**Versions**

+ Chromium: Chromium 86.0.4240.111
+ Image: zenika/alpine-chrome@sha256:2847c4e24af279ee12eea011ba9f669dd92adf317d01fcf3a7a51f88f8a8214b
Docker Engine: 20.10.1
+ Puppeteer: 5.5.0

**Additional context**
This happens very rarely when running the program locally. When running through the image, this happens every time.

### PR DESCRIPTION
**Pull request by zigarn**
Fixes #152
Supersedes #182

DockerHub autobuild doesn't support build for ARM (docker/roadmap#109), so switch to GitHub Actions for multi-platform (amd64 & arm64) docker build.

Build amd64 & test for every PR.
Build multi-platform & push only when push on `master`.

Adaptations:
 - sequential build of flavors: this way the `with-xxx` is built against the `latest` from same build instead of previous one
 - added tests for each flavor (requires absolute paths in Dockerfiles)

@jlandure: I did put my own `DOCKERHUB_(USERNAME|TOKEN)` as secret, should change it for the technical account [dsizenika](https://hub.docker.com/u/dsizenika).

Result: https://github.com/zigarn/alpine-chrome/actions/runs/2327992742 & https://hub.docker.com/repository/docker/zigarn/alpine-chrome-multi-arch/tags

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,185 @@
+---
+
+name: build
+on:
+  push:
+    branches:
+      - main
```
Comment (hgwood) [inline .github/workflows/build.yml:7]: I guess this should say master?

**Review by hgwood — COMMENTED** [review]
Thanks a lot!

File: .github/workflows/build.yml (hunk 1 above)
Comment (zigarn) [reply .github/workflows/build.yml:7]: Good point!

Comment (zigarn) [reply .github/workflows/build.yml]: Fixed.

**Comment by willjosefi** [comment]
hey guys!
seems like the PR is merged but I'm not seeing the images from arm64 inside docker hub
+ seems like the gh actions had a failure

**Comment by zigarn** [comment]
Yes, there are some authentication issues blocking the push.
@jlandure or @hgwood: if you can give me back access to the Zenika's org on DockerHub, I can get the GCR credentials from there (if it's still there) and test what's wrong with it.

**Comment by hgwood** [comment]
@zigarn Done :)

**Comment by zigarn** [comment]
Thanks!
Looks like GCR does not accept `_json_key_base64` username (https://cloud.google.com/container-registry/docs/advanced-authentication#json-key), unlike GAR (https://cloud.google.com/artifact-registry/docs/docker/authentication#json-key) and despite the doc of `docker-login` action (https://github.com/marketplace/actions/docker-login#service-account-based-authentication)

**Comment by zigarn** [comment]
Improved by using a not base64 version of the GCR_JSON_KEY: login is OK: https://github.com/Zenika/alpine-chrome/runs/6858273478?check_suite_focus=true
But push is not...

**Comment by zigarn** [comment]
@willjosefi: arm64 build are now available (for base, with-node, with-deno & with-chromedriver at least)! Still some issues with other tests (that I don't understand as it's working locally and it was working in my repo https://github.com/zigarn/alpine-chrome/runs/6442494884?check_suite_focus=true)

**Comment by justinh998** [comment]
Hi why are there no up to date arm64 images?
The Link : https://github.com/zigarn/alpine-chrome/actions/runs/2327992742 doesnt work.

**Comment by justinh998** [comment]
> @willjosefi: arm64 images are now available! (https://github.com/zigarn/alpine-chrome/actions/runs/2327992742)

Hi why are there no up to date arm64 images?
The Link : https://github.com/zigarn/alpine-chrome/actions/runs/2327992742 doesnt work.

**Comment by zigarn** [comment]
> > @willjosefi: arm64 images are now available! (https://github.com/zigarn/alpine-chrome/actions/runs/2327992742)
> 
> Hi why are there no up to date arm64 images? The Link : https://github.com/zigarn/alpine-chrome/actions/runs/2327992742 doesnt work.

This was the link on my fork.
The build is going well on the main repository: https://github.com/Zenika/alpine-chrome/actions/runs/11155899792
You can see that latest images published on Docker Hub 5 days ago are with arm64 version: https://hub.docker.com/r/zenika/alpine-chrome/tags

**Pull request merged**